babel.config.js

```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  presets: [require.resolve("@babel/preset-typescript")],
+};
```

Comment:
```suggestion
module.exports = {
  presets: [
    require.resolve("@babel/preset-react"),
    [
      require.resolve("@babel/preset-typescript"),
      { allExtensions: true, disallowAmbiguousJSXLike: true, isTSX: true },
    ],
  ],
};
```

+ install `@babel/preset-react`. I think that better code frames should be thrown here. Sometimes they are... when the parsing error happens in Babel but here it sometimes was happening in Rollup itself and that didn't produce helpful code frames. Gonna investigate what can be done about it

Reply:
Note that not all of those options passed to the TS preset might actually be needed, I just hacked it around quickly and didn't recheck back.

Reply:
Unfortunately - you might not be able to use Preconstruct here at all because you are importing `.css` files. That's not quite standard... so there is no good answer as to what should actually be done here. We could add an experimental/unsafe option to just emit/preserve those as assets but once you ship a library like this... it won't work without a bundler.

Reply:
I was just trying this locally. I _think_ the parsing error is related to CSS modules.

Reply:
Yup! I needed to also add babel-plugin-css-modules-transform.